src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageLoader.cs

```diff
@@ -59,14 +69,45 @@ public CompiledPageActionDescriptor Load(PageActionDescriptor actionDescriptor)
                 handlerMethods = CreateHandlerMethods(pageType);
             }
 
+            // Filters are only allowed to be declared on an explicitly specified model.
+            var filters = actionDescriptor.FilterDescriptors;
+            if (modelType != null && modelType != pageType)
+            {
+                var modelAttributes = modelType.GetCustomAttributes(inherit: true);
```

Comment:
Do we want PageModel to implement `IPageFilter, IAsyncPageFilter` and discover those too?